### ISSUE TITLE
New OT: Position#_getTransformedByMove is incorrect for "non-moves"

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5#4396]
In some cases current implementation of `Position#_getTransformedByMove` returns wrong values. It happens when the transformed position is the same as passed `sourcePosition` and the `targetPosition` is at the end of the moved range or at the beginning of the moved range (`sourcePosition`) so it is not really any move.

```html
<p>F^[oo]bar</p>
```

`[]` is moved to the position `^` or `]`.

What happens is: `targetPosition` after transforming by `oo` remove is at the same place as `^`. Since we don't have `insertBefore` flag anymore, `^` position is transformed like the nodes were moved before it.

So basically after this transformation the position goes to:

```html
<p>F[oo]^bar</p>
```

Which is incorrect because `oo` didn't really move.
